src/dist/cluster_scheduler/scheduler_providers/docker_scheduler.cpp

```diff
@@ -160,6 +140,47 @@ error_code docker_scheduler::initialize()
     return ::dsn::ERR_OK;
 }
 
+void docker_scheduler::get_app_list(std::string& ldir,/*out*/std::vector<std::string>& app_list )
+{
+#ifndef _WIN32
+    std::string popen_command = "cat " + ldir + "/applist";
+    FILE *f = popen(popen_command.c_str(),"r");
+    char buffer[128];
+    fgets(buffer,128,f);
+    ::dsn::utils::split_args(buffer, app_list, ' ');
+#endif
+}
+
+
+void docker_scheduler::write_machine_list(std::string& name, std::string& ldir)
+{
+    std::vector<std::string> app_list;
+
+    get_app_list(ldir,app_list);
+
+    for( auto& app : app_list)
+    {
+        std::string machine_file = ldir + "/" + app + "list";
+        std::vector<std::string> machine_list;
+        std::vector<std::string> f_list;
+        std::vector<std::string> a_list;
+        int count = 1;
+        _mgr.get_machine( count, f_list, a_list);
+        _machine_map[name].insert(_machine_map[name].begin(),a_list.begin(),a_list.end());
+        for( auto& machine: a_list )
+        {
+            std::string command = "echo "+ machine +">> " + machine_file; 
+            system(command.c_str());
```

Comment:
Should use std::ofstream etc. to make it more reliable. We should use ```system``` etc. only when we don't have other better approaches.